### PR TITLE
Feature/signin with phonenumber

### PR DESCRIPTION
**Pull request by Eimen2018**
- adds a method to sign in with phone number 
- adds a method to verifyOtp

### CODE REVIEW

File: packages/stacked_firebase_auth/lib/src/firebase_authentication_service.dart [hunk 1]
```diff
@@ -317,6 +317,23 @@ class FirebaseAuthenticationService {
     );
   }
 
+    /// Phone Number Login
+  Future<ConfirmationResult> signInWithPhoneNumber(String phoneNumber) async {
+    return firebaseAuth.signInWithPhoneNumber(phoneNumber);
```
Comment (eb-so) [inline packages/stacked_firebase_auth/lib/src/firebase_authentication_service.dart]: You should add a try/catch the same you did below

File: packages/stacked_firebase_auth/lib/src/firebase_authentication_service.dart [hunk 2]
```diff
@@ -317,6 +317,23 @@ class FirebaseAuthenticationService {
...
+  }
+
+  Future<FirebaseAuthenticationResult> verifyOtp(
+      ConfirmationResult confirmationResult, String otp) async {
+    try {
+      UserCredential userCredential = await confirmationResult.confirm(otp);
+      return FirebaseAuthenticationResult(user: userCredential.user);
+    } catch (e) {}
+    return FirebaseAuthenticationResult.error(
```
Comment (eb-so) [inline packages/stacked_firebase_auth/lib/src/firebase_authentication_service.dart]: Place it inside the catch closer and add the error to the end of the error message

Comment (Eimen2018) [reply packages/stacked_firebase_auth/lib/src/firebase_authentication_service.dart]: @eb-so I thought i did 😆 ... my bad

**Comment by eb-so** [comment]
LGTM

**Review by FilledStacks — APPROVED** [review]
Wooohooo! LGTM

**Pull request merged**